From fa66089bd5401193b166fc2cb567626e62dafd7a Mon Sep 17 00:00:00 2001
From: Andreas Ternstedt <a.ternstedt@setek.se>
Date: Tue, 28 Oct 2025 14:43:34 +0000
Subject: [PATCH] mlas: Remove ARMv8.2+ optimizations for Cortex-A53
 compatibility

Remove dependancies of ARMv8.2-a instruction set.
These are optional runtime optimizations not supported by Cortex-A53 (ARMv8.0).

Signed-off-by: Andreas Ternstedt <a.ternstedt@setek.se>

---
 cmake/onnxruntime_mlas.cmake     | 14 --------------
 onnxruntime/core/mlas/inc/mlas.h |  2 +-
 2 files changed, 1 insertion(+), 15 deletions(-)

diff --git a/cmake/onnxruntime_mlas.cmake b/cmake/onnxruntime_mlas.cmake
index e0ccc504d..79293a9f0 100644
--- a/cmake/onnxruntime_mlas.cmake
+++ b/cmake/onnxruntime_mlas.cmake
@@ -335,20 +335,6 @@ else()
           ${MLAS_SRC_DIR}/qgemm_kernel_udot.cpp
           ${MLAS_SRC_DIR}/qgemm_kernel_sdot.cpp
         )
-        if (NOT APPLE)
-          set(mlas_platform_srcs
-            ${mlas_platform_srcs}
-            ${MLAS_SRC_DIR}/aarch64/HalfGemmKernelNeon.S
-            ${MLAS_SRC_DIR}/activate_fp16.cpp
-            ${MLAS_SRC_DIR}/dwconv.cpp
-            ${MLAS_SRC_DIR}/halfgemm_kernel_neon.cpp
-            ${MLAS_SRC_DIR}/pooling_fp16.cpp
-          )
-          set_source_files_properties(${MLAS_SRC_DIR}/aarch64/HalfGemmKernelNeon.S PROPERTIES COMPILE_FLAGS " -march=armv8.2-a+fp16 ")
-          set_source_files_properties(${MLAS_SRC_DIR}/activate_fp16.cpp PROPERTIES COMPILE_FLAGS " -march=armv8.2-a+fp16 ")
-          set_source_files_properties(${MLAS_SRC_DIR}/dwconv.cpp PROPERTIES COMPILE_FLAGS " -march=armv8.2-a+fp16 ")
-          set_source_files_properties(${MLAS_SRC_DIR}/pooling_fp16.cpp PROPERTIES COMPILE_FLAGS " -march=armv8.2-a+fp16 ")
-        endif()
 
         if(ONNXRUNTIME_MLAS_MULTI_ARCH)
             onnxruntime_add_static_library(onnxruntime_mlas_arm64 ${mlas_platform_srcs})
diff --git a/onnxruntime/core/mlas/inc/mlas.h b/onnxruntime/core/mlas/inc/mlas.h
index fd6b3df93..8a0dd8d19 100644
--- a/onnxruntime/core/mlas/inc/mlas.h
+++ b/onnxruntime/core/mlas/inc/mlas.h
@@ -85,7 +85,7 @@ Abstract:
 // When building an universial binary for APPLE, this flag would
 // cause trouble for x64 target.
 
-#define MLAS_F16VEC_INTRINSICS_SUPPORTED
+//#define MLAS_F16VEC_INTRINSICS_SUPPORTED
 
 #endif // 
 #endif // ARM64
-- 
2.34.1
